enhancements/machine-config/manage-boot-images.md

```diff
@@ -282,44 +285,108 @@ spec:
 ```
 #### Tracking boot image history
 
-This proposal will also introduce a new CR, `MachineSetBootImageHistory` for tracking boot image history. As a starting point, here is a stub type definition for this:
+This proposal will also introduce a new CR, `BootImageHistory` for tracking boot image history. As a starting point, here is a stub type definition for this:
 
 ```
-type MachineSetBootImageHistory struct {
+type BootImageHistory struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   MachineSetBootImageHistorySpec   `json:"spec,omitempty"`
-	Status MachineSetBootImageHistoryStatus `json:"status,omitempty"`
+	Spec   BootImageHistorySpec   `json:"spec,omitempty"`
+	Status BootImageHistoryStatus `json:"status,omitempty"`
+}
+
+// BootImageHistorySpec defines the desired state of BootImageHistory
+type BootImageHistorySpec struct {
 }
 
-// MachineSetBootImageHistorySpec defines the desired state of MachineSetBootImageHistory
-type MachineSetBootImageHistorySpec struct {
-	MachineSetName string                   `json:"machineSetName"`
-	Details        []BootImageHistoryDetail `json:"details"`
+// BootImageHistoryStatus defines the observed state of BootImageHistory
+type BootImageHistoryStatus struct {
+	// machineResourceReference contains identifying information of the machine management resource being tracked.
+	// +kubebuilder:validation:Required
+	// +required
+	MachineResourceReference MachineResourceReference `json:"machineResourceReference"`
+	// details is a list of boot image history entries of the machine resource.
+	// +optional
+	Details []BootImageHistoryDetail `json:"details,omitempty"`
 }
 
-// MachineSetBootImageHistoryStatus defines the observed state of MachineSetBootImageHistory
-type MachineSetBootImageHistoryStatus struct {
+type MachineResourceReference struct {
+	// name is the machine management resource's name
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// kind is the machine management resource's kind
+	// +kubebuilder:validation:Required
+	// +required
+	Kind string `json:"kind"`
```

Comment:
Should use `Resource` not `Kind`. `Resource` can be mapped to `Kind` in all cases, the opposite is not true and you need `Resource` to be able to form API requests. `Machine` would be a `Kind`, where `machines` would be the `Resource`